retrofit/src/main/java/retrofit/MethodInvoker.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit;
+
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.ResponseBody;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import retrofit.http.Streaming;
+
+import static retrofit.Utils.methodError;
+
+final class MethodInvoker<T> {
```

Comment:
It's definitely not a `MethodInvoker`, that's a thing that calls methods, not a thing that responds to method calls